pkg/grpc/grpc.go

```diff
@@ -207,29 +203,7 @@ func (g *Manager) connectRemote(
 		)
 	}
 
-	if g.auth != nil {
-		signedCert := g.auth.GetCurrentSignedCert()
-		var cert tls.Certificate
-		cert, err = tls.X509KeyPair(signedCert.WorkloadCert, signedCert.PrivateKeyPem)
-		if err != nil {
-			return nil, fmt.Errorf("error loading x509 Key Pair: %w", err)
-		}
-
-		var serverName string
-		if id != "cluster.local" {
-			serverName = id + "." + namespace + ".svc.cluster.local"
-		}
-
-		//nolint:gosec
-		ta := credentials.NewTLS(&tls.Config{
-			ServerName:   serverName,
-			Certificates: []tls.Certificate{cert},
-			RootCAs:      signedCert.TrustChain,
-		})
-		opts = append(opts, grpc.WithTransportCredentials(ta))
-	} else {
-		opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
-	}
+	opts = append(opts, g.sec.GRPCDialOptionUnknownTrustDomain(namespace, id))
```

Comment:
Would this cause issues if a sidecar was trying to connect to a v1.11 sidecar?

Reply:
Yes, we are missing a legacy wrapper for `GRPCDialOptionUnknownTrustDomain`.